11_FFTBloom/main.cpp

```diff
@@ -431,8 +454,9 @@ class FFTBloomApp final : public application_templates::MonoDeviceApplication, p
 			deviceLocalBufferParams.queueFamilyIndices = &queueFamilyIndex;
 			// Axis on which we perform first FFT is the only one that needs to be padded - in this case it's the y-axis
 			uint16_t firstAxis = 1;
-			deviceLocalBufferParams.size = getOutputBufferSize(marginSrcDim, 3, {&firstAxis, 1}, true, useHalfFloats);
-			deviceLocalBufferParams.usage = nbl::asset::IBuffer::E_USAGE_FLAGS::EUF_STORAGE_BUFFER_BIT | nbl::asset::IBuffer::E_USAGE_FLAGS::EUF_SHADER_DEVICE_ADDRESS_BIT;
+			uint32_t3 paddedSrcDimensions = { m_marginSrcDim.width, m_marginSrcDim.height, m_marginSrcDim.depth };
+			deviceLocalBufferParams.size = fft::getOutputBufferSize(paddedSrcDimensions, 3, {&firstAxis, 1}, true, m_useHalfFloats);
```

Comment:
do all 2 (or 3) extents need to be paddeed in your intermediate buffer?

I think not, because first you do a bunch of Y-axis (column) FFTs, then you do X-axis but you don't really store them anywhere, just multiply spectra, do  inverse fft and store back to same indices immediately